Makefile

```diff
@@ -635,10 +638,73 @@ verify-links:
 	fi;
 .PHONY: verify-links
 
-verify: verify-codegen verify-crds verify-helm-schemas verify-helm-charts verify-deploy verify-lint verify-helm-lint verify-links verify-examples verify-docs-api verify-monitoring
+# $1 - path to the target being patched
+# $2 - glob pattern for patch files
+define patch-bundle
+	for f in $(2); do \
+		$(YQ) -i -e '. *= load("'"$$f"'")' $(1); \
+	done
+endef
+
+# $1 - path to manifests directory
+define fix-bundle-manifests-filenames
+	find $(1) -type f -name '*:*' | while read file; do \
+		newfile=$$(echo "$$file" | sed 's/:/-/g'); \
+		mv "$$file" "$$newfile"; \
+	done
+endef
+
+# $1 - logo path
+# $2 - logo patch path
+define update-bundle-patches-logo
+	$(YQ) -i ".spec.icon[0].base64data = \"$$( base64 -w 0 '$(1)' )\"" '$(2)'
+	$(YQ) -i ".spec.icon[0].mediatype = \"$$( file --mime-type -b '$(1)' )\"" '$(2)'
+endef
+
+# $1 - metadata file
+# $2 - patch file
+define update-bundle-patches-versions
+	$(YQ) eval-all -i -P '\
+	select(fi==0).spec.minKubeVersion = ( select(fi==1) | .operator.minKubernetesVersion ) | \
+	select(fi==0).metadata.annotations."com.redhat.openshift.versions" = ( select(fi==1) | "v" + .operator.minOpenShiftVersion + "-v" + .operator.maxOpenShiftVersion ) | \
+	select(fi==0)' \
+	'$(2)' '$(1)'
+endef
+
+# $1 - bundle path
+define update-bundle-patches
+	$(call update-bundle-patches-logo,./logo.svg,$(1)/patches/logo.clusterserviceversion.yaml)
+	$(call update-bundle-patches-versions,assets/metadata/metadata.yaml,$(1)/patches/versions.clusterserviceversion.yaml)
```

Comment:
I think the two files that are updated automatically could still use a comment saying this, otherwise it can only be discovered by going through the Makefile targets. But it can be done separately, without retriggering CI.